dream_layer_backend/txt2img_workflow.py

```
@@ -318,126 +318,226 @@ def inject_controlnet_parameters(workflow, controlnet_data):
+            original_positive = None
+            original_negative = None
+            for node_id, node_data in prompt.items():
+                if node_data.get('class_type') == 'CLIPTextEncode':
+                    inputs = node_data.get('inputs', {})
+                    if 'positive' in inputs.get('text', '').lower() or node_id == '4':
+                        original_positive = node_id
+                    elif 'negative' in inputs.get('text', '').lower() or node_id == '5':
+                        original_negative = node_id
+            
+            # Get next available node ID
+            max_node_id = max([float(k) for k in prompt.keys() if k.replace('.', '').isdigit()])
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)